CMakeLists.txt

```diff
@@ -183,8 +183,9 @@ if(NO_STACK_PROTECTOR)
     target_compile_options(${PROJECT_NAME} PRIVATE -Wstack-protector -fstack-protector-all)
 endif()
 
-if(S2N_UNSAFE_FUZZING_MODE)
-    target_compile_options(${PROJECT_NAME} PRIVATE -fsanitize-coverage=trace-pc-guard -fsanitize=address,undefined,leak -fuse-ld=gold -DS2N_ADDRESS_SANITIZER=1)
+if(S2N_FUZZ_TEST)
+    target_compile_options(${PROJECT_NAME} PRIVATE -fsanitize-coverage=trace-pc-guard -fsanitize=leak)
+    target_link_libraries(${PROJECT_NAME} PRIVATE -fsanitize-coverage=trace-pc-guard -fsanitize=leak)
```

Comment:
```suggestion
    target_compile_options(${PROJECT_NAME} PRIVATE -fsanitize-coverage=trace-pc-guard)
    target_link_libraries(${PROJECT_NAME} PRIVATE -fsanitize-coverage=trace-pc-guard)
```
I think currently we're running without the leak sanitizer?

Also, if we want to be pedantic I think the more "cmake-y" way to add these options would be to add them as`PUBLIC` options to the test executables. Then the project should just inherit the link/compile options.

Reply:
Removed the leak option and make compile/link option PUBLIC in the latest commit

Reply:
Hmm, then let's maybe revisit this once we re-enable the asan/leak detection on all of the fuzz tests? What you have should be fine for now but let's make it a bit more "cmake-y" on the next PR.

My preference with CMake is to only toggle options on the "most appropriate" target, and then rely on PRIVATE/PUBLIC to get the correct setup with the rest of the build.

E.g. My mental model is that we turn leak detection on for the fuzz tests, which just happens to require the flags also be applied to the library. So they way I'd vote to do this is
```
target_compile_options(fuzz_test PUBLIC -fsanaitize=leak)
```
The public flag here means that anything that links against the fuzz test will inherit the compile options (`-fsanitize=leak`) including the s2n-tls library.